src/Universum/Monoid.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE CPP #-}
+
+module Universum.Monoid
+       (
+#if ( __GLASGOW_HASKELL__ >= 800 )
+         module Data.List.NonEmpty
+       , module Data.Monoid
+       , module Data.Semigroup
+#else
+         module Data.Monoid
+#endif
+       , maybeToMonoid
+       ) where
+
+#if ( __GLASGOW_HASKELL__ >= 800 )
+import Data.List.NonEmpty (NonEmpty (..), nonEmpty)
+import Data.Monoid hiding ((<>))
```

Comment:
Could you please specify here explicitly what we export from `Data.Monoid` module?

You can found this information in this module:

https://www.stackage.org/haddock/lts-9.18/base-4.9.1.0/Data-Monoid.html

Just for visibility :) I think it's better to see what exactly module reexports. Hopefully, `Data.Monoid` doesn't have a lot of things... For `Data.Semigroup` we specified everything explicitly. Also, You will have same `Data.Monoid` under both `#if-else-endif` branches. Because of that you can move this import out of `#if-else-endif` branches.